### ISSUE TITLE
(Item Challenge) Track per player

### ISSUE THREAD

**Issue by dvdcd** [Plagiatus/datapacks#2]
I'd like to use this on a multiplayer server as a sort of race to the finish. at the moment i can't because the challenge is tracked globally.

**Comment by Plagiatus** [comment]
The problem with this is, that you cannot know which player actually put in which item into the chest, making the correct detection basically impossible.
I may be able to set something up that allows you to have one shrine per person, would that work for you?

**Comment by dvdcd** [comment]
Yeah! another idea is to have a modular shrine with multiple chests inside or something. Or even better (i'm just brainstorming at this point) Can you do some sort of test for the player that's closest to the chest and have that player unlock the advancement? (Thanks for the fast response btw)

**Comment by Plagiatus** [comment]
the "closest player" thing was the obvious first approach to this, but what if multiple people are close to the shrine at the same time? you can't rely on players not trying to get closer to the chest to steal opponents points tbh.
but you could modify the shrine with some redstone so only one person at a time can be inside? 🤔

**Comment by dvdcd** [comment]
Yeah, so can you test if a player is inside the shrine and then place a redstone block under a iron door to lock it?
